sources/packages/backend/apps/api/src/auth/guards/authorized-parties.guard.ts

```diff
@@ -54,23 +53,23 @@ export class AuthorizedPartiesGuard implements CanActivate {
    * Determines if the client is authorized through the expected IDP.
    * @param authorizedParty authorized party type to be checked.
    * @param idp identity provider used for authentication on Keycloak.
-   * @returns true if IDP is allowed, otherwise, false.
+   * @returns true if the identity provider is allowed, otherwise, false.
    */
   private isAllowedIDP(
     authorizedParty: AuthorizedParties,
-    idp: IdentityProviders,
+    identityProvider: IdentityProviders,
   ): boolean {
     switch (authorizedParty) {
       case AuthorizedParties.student:
-        return [IdentityProviders.BCeID, IdentityProviders.BCSC].includes(idp);
+        return [IdentityProviders.BCeIDBoth, IdentityProviders.BCSC].includes(
```

Comment:
What happens here when a business bceid user login to student portal? should we use evaluateSpecificIdentityProvider to check?

Reply:
Right now there is nothing preventing a business BCeID from login into the portal and there is no requirement to prevent it.
@JasonCTang it is something that the Ministry would be interested in?